docs(material/stepper): expand a11y section (#23617)
angular/components: src/material/stepper/stepper.md

step content is automatically set based on step selection change.

The stepper and each step should be given a meaningful label via `aria-label` or `aria-labelledby`.

> When a stepper is used on a smaller screen, verical steppers are preferred over

Reviewer: verical -> vertical

Reviewer 2: ```markdown
Prefer vertical steppers when building for small screen sizes, as horizontal steppers typically take up significantly more horizontal space, thus introducing horizontal scrolling. Applications with multiple scrolling dimensions make content harder to consume for some users. See the [Responsive Stepper section](#responsive-stepper) above for an example on building a stepper that adjusts its layout based on viewport size.
```
* Eliminates passive voice (e.g. "stepper is used", "steppers are preferred")
* Adds some context on _why_ folks would want to be careful with horizontal steppers on small screens
* Links to the example above

> horizontal steppers because the text in the header will be more visible.

Reviewer: Might be worth mentioning here that there's an example of a responsive stepper further up in the doc.

#### Forms

> Steppers often have forms inside of it. It's best practice for form fields to

Reviewer: inside of it -> inside of them

Reviewer 2: ```suggestion
Steppers often contain forms and form controls. If validation errors inside of a stepper's form prevents moving to another step, make sure that your form controls communicate error messages to assistive technology. This helps the user know why they can't advance to another step. You can accomplish this by using `<mat-error>` with `<mat-form-field>`, or by using an ARIA live region. 
```
* Changes for passive voice and grammar 
* Clarify that `<mat-error>` is one possible way to communicate updates

> have `<mat-error>` for fields that can be incorrect and throw an error.

Reviewer: nit: I'm not sure that "throw an error" is the correct terminology here since it sounds like there was an actual runtime error, whereas these are validation errors.

This allows for the error message to be conveyed to users who use screen
readers. Otherwise for a linear stepper, a screen reader user can try to
navigate to the next step but if there was an error they won't be able to and
then must look for where the error came from.

> When you navigate away from a step with an error in a non-linear stepper, the
> error message appears in the step header. Announcing the error with
> `liveAnnouncer` will help screen reader users know that the step they just
> navigated away from has an error.

Reviewer 2: ```suggestion
When a step contains a forms validation error, `MatStepper` will display the error in the step's header. For non-linear steppers, you should use an ARIA live region to announce error messages when users navigate away from a step with an error message. 
```

 Is this accurate to what you're trying to get across?

step content is automatically set based on step selection change.

The stepper and each step should be given a meaningful label via `aria-label` or `aria-labelledby`.

Prefer verticl steppers when building for small screen sizes, as horizontal

> steppers typically take up significantly more horizontal space thus introducing

Reviewer 2: ```suggestion
steppers typically take up significantly more horizontal space thus introduce
```
